target_chains/fuel/contracts/pyth-contract/src/main.sw

```diff
@@ -848,57 +847,62 @@ impl PythGovernance for Contract {
-            GovernanceAction::SetFee => {
-                let sf = GovernanceInstruction::parse_set_fee_payload(gi.payload);
-                log(sf);
-                set_fee(sf);
-            },
-            GovernanceAction::SetValidPeriod => {
-                let svp = GovernanceInstruction::parse_set_valid_period_payload(gi.payload);
-                log(svp);
-                set_valid_period(svp);
-            },
-            GovernanceAction::RequestGovernanceDataSourceTransfer => {
-                // RequestGovernanceDataSourceTransfer can be only part of AuthorizeGovernanceDataSourceTransfer message
-                // The `revert` function only accepts u64, so as
-                // a workaround we use require.
-                require(false, PythError::InvalidGovernanceMessage);
-            },
-            _ => {
-                // The `revert` function only accepts u64, so as
-                // a workaround we use require.
-                require(false, PythError::InvalidGovernanceMessage);
-            }
+    require(
+        gi.target_chain_id == chain_id() || gi.target_chain_id == 0,
+        PythError::InvalidGovernanceTarget,
+    );
+
+    match gi.action {
+        GovernanceAction::UpgradeContract => {
+            require(gi.target_chain_id != 0, PythError::InvalidGovernanceTarget);
+            // TODO: implement upgrade_upgradeable_contract(uc) when Fuel releases the upgrade standard library;
+            log("Upgrade functionality not implemented");
+            revert(0u64);
+        },
+        GovernanceAction::AuthorizeGovernanceDataSourceTransfer => {
+            let agdst = GovernanceInstruction::parse_authorize_governance_data_source_transfer_payload(gi.payload);
+            log(agdst);
+            authorize_governance_data_source_transfer(agdst);
+        },
+        GovernanceAction::SetDataSources => {
+            let sdsp = GovernanceInstruction::parse_set_data_sources_payload(gi.payload);
+            log(sdsp);
+            set_data_sources(sdsp);
+        },
+        GovernanceAction::SetFee => {
+            let sf = GovernanceInstruction::parse_set_fee_payload(gi.payload);
+            log(sf);
+            set_fee(sf);
+        },
+        GovernanceAction::SetValidPeriod => {
+            let svp = GovernanceInstruction::parse_set_valid_period_payload(gi.payload);
+            log(svp);
+            set_valid_period(svp);
+        },
+        GovernanceAction::RequestGovernanceDataSourceTransfer => {
+            // RequestGovernanceDataSourceTransfer can be only part of AuthorizeGovernanceDataSourceTransfer message
+            // The `revert` function only accepts u64, so as
+            // a workaround we use require.
+            require(false, PythError::InvalidGovernanceMessage);
+        },
+        _ => {
+            // The `revert` function only accepts u64, so as
+            // a workaround we use require.
+            require(false, PythError::InvalidGovernanceMessage);
```

Comment:
there is a compiler issue with fuel where it fails to compile with `Too many arguments, cannot handle.: Immediate12TooLarge` error and the solution was to make the contract functions call internal functions